xbmc/application/ApplicationPlayer.cpp

```diff
@@ -473,25 +472,34 @@ int CApplicationPlayer::GetSubtitle()
     return 0;
 }
 
-bool CApplicationPlayer::GetSubtitleVisible()
+bool CApplicationPlayer::GetSubtitleVisible() const
 {
-  std::shared_ptr<IPlayer> player = GetInternal();
-  return (player && player->GetSubtitleVisible());
+  const auto player = GetInternal();
+  return player && player->GetSubtitleVisible();
 }
 
-bool CApplicationPlayer::CanPause()
+bool CApplicationPlayer::CanPause() const
 {
-  std::shared_ptr<IPlayer> player = GetInternal();
+  const auto player = GetInternal();
   return (player && player->CanPause());
 }
 
+bool CApplicationPlayer::HasTeletextCache() const
+{
+  const auto player = GetInternal();
+  if (player)
+    return player->HasTeletextCache();
+  else
+    return false;
```

Comment:
that is a new warning i guess
```suggestion
  if (player)
    return player->HasTeletextCache();

  return false;
```

Reply:
uhm what warning would that avoid?

Reply:
warning: control reaches end of non-void function.
honestly that warning is nonsense imo but i got it somewhere previously
jenkins seems to be quiet about it now 🤔

Reply:
it either takes the true branch or the else branch. both return. there is no reaching any end of any function here.